index.html

```diff
@@ -702,27 +712,150 @@ <h3>Validating an ExposedThingInit</h3>
+        </li>
+        <li>
+          If directory discovery is not supported by the implementation, reject |promise|
+          with {{NotSupportedError}} and abort these steps.
+        </li>
+        <li>
+          Let |discovery:ThingDiscoveryProcess| be a new {{ThingDiscoveryProcess}} object.
+        </li>
+        <li>
+          Set |discovery|'s {{ThingDiscoveryProcess/[[url]]}} to |url:USVString|.
+        </li>
+        <li>
+          Set |discovery|'s {{ThingDiscoveryProcess/[[filter]]}} to |filter:ThingFilter|.
+        </li>
+        <li>
+          Request the underlying platform to start the directory discovery process.
+          <p class="note">
+            This is a placeholder for more details in the discovery algorithm.
+            Implementations should follow the procedures described in the
+            [[WOT-DISCOVERY]] and [WOT-PROTOCOL-BINDINGS] specifications.
+            Some normative steps are indicated below.
+          </p>
+          <ol>
+            <li>
+              If |url| is not a <a>TD Directory</a> or if the underlying
+              implementation cannot support the <a>Protocol Binding</a>
+              indicated by |url|, reject |promise|
+              with {{NotSupportedError}} and terminate these steps.
+            </li>
+            <li>
+              If |filter| is not supported by the implementation, reject |promise|
+              with {{NotSupportedError}} and terminate these steps.
+            </li>
+            <li>
+              Run the <a>discovery process</a> given |discovery|.
+              <p class="note">
+                From this point on, errors are recorded only on
+                {{ThingDiscoveryProcess/error}}, but don't affect |promise| any longer.
+              </p>
+            </li>
+          </ol>
+        </li>
+        <li>
+          Resolve |promise| with |discovery|.
+        </li>
+      </ol>
+    </div>
+  </section>
+
+  <section> <h3>The <dfn>requestThingDescription()</dfn> method</h3>
+    <pre class="idl">
+      partial namespace WOT {
+        Promise&lt;ThingDescription&gt; requestThingDescription(USVString url);
+      };
+    </pre>
+    <div>
+      Belongs to the <a>WoT Discovery</a> conformance class.
+      Requests a <a>Thing Description</a> from the given URL.
+      The method MUST run the following steps:
+      <ol>
+        <li>
+          Return a {{Promise}} |promise:Promise| and execute the next steps [=in parallel=].
+        </li>
+        <li>
+          If invoking this method is not allowed for the current scripting
+          context for security reasons, reject |promise| with a {{SecurityError}}
+          and abort these steps.
+        </li>
+        <li>
+          If fetching a <a>Thing Description</a> is not supported by the
+          implementation, reject |promise| with {{NotSupportedError}} and
+          abort these steps.
```

Comment:
Here is also a good place to put the same error condition as specified in the `exploreDirectory` method:
> if the underlying
              implementation cannot support the <a>Protocol Binding</a>
              indicated by |url|

Reply:
Mmm, in this case I think a `NotSupportedError` is straightforward, but we can discuss it.

Reply:
From 19/12/2022:
- @zolkis if we go into detail there would be a lot of other clauses, it is better to stay generic.